notebooks/test_idata_general_time_representation.py

```diff
@@ -0,0 +1,241 @@
+"""
+Experiment file for testing
+general representation of timepoints
+in idata object (intervals not just
+1D).
+"""
+
+# %% LIBRARY IMPORTS
+
+from datetime import datetime, timedelta
+
+import arviz as az
+import polars as pl
+
+import forecasttools
+
+# %% LOAD IDATA WO DATES
+
+idata_wo_dates = forecasttools.nhsn_flu_forecast_wo_dates
+
+print(idata_wo_dates["observed_data"]["obs_dim_0"])
+
+# NOTE: assumptions about different chains
+# ought to be taken into account, not currently
+# taken into account; might matter.
+
+# %% OPTION 1 FOR ADDING DATES
+
+
+def option_1_add_dates_as_coords_to_idata(
+    idata_wo_dates: az.InferenceData,
+    dim_name: str,
+    group_date_mapping: dict[str, tuple[str, timedelta]],
```

Comment:
See https://github.com/CDCgov/forecasttools-py/pull/35#issuecomment-2448190737. I think all your uses of "group" are actually "variables"?

Reply:
I feel that sections such as the following distinguish between groups and dimensions fairly clearly: 

```python
if dim_name not in idata_group.dims:
    print(
        f"Warning: Dimension '{dim_name}' not found in group '{group_name}'."
    )
    continue
```

Reply:
But not all of the variables in a group should be given the same coordinates. How do you specify the variable you want to modify?

Reply:
> But not all of the variables in a group should be given the same coordinates. How do you specify the variable you want to modify?

I think this is a useful remark. Imagine having a separate `xarray` DataSet for weekly Rt values along with an `idata` object that has the group `posterior_predictive` with daily hospital admissions stored in a variable called `obs`. I can imagine a situation where one adds the Rt values dataset as a new group to the `idata`. However, I see now that one might have in `posterior_predictive` another variable `Rt` with different coordinates. My current approach regards to the (lower likelihood) situation where another variable of interest has been added via a new group. My approach does not suit multiple variables of interest within the same group which use different coordinates.